### DISCUSSION TITLE
July Roadmap 2022

### DISCUSSION THREAD

**Discussion by kylecarbs** [coder/coder#3325] [RFCs]
Update 7/29/2022: Unfinished items have been carried over and re-prioritized in the August Roadmap https://github.com/coder/coder/issues/3182

This is not a hard commitment to all of the items presented and other items not listed may also be worked on. This is an experiment in having a public roadmap. If it works well, we may iterate and continue having a monthly roadmap issue. 

### ✏️ template.yaml (postponed)
Template name, description, and schedule are not stored in a template version. There is no way to source control these properties, and it’s likely to become problematic as we create more.

https://github.com/coder/coder/issues/2987 @sreya 

### 📜 Resource Metadata + Secrets

Developers want to see additional metadata about resources in Coder. This could be an instance type, the number of cores an instance has, or a link to the cloud resource itself (say in a GCP dashboard).

https://github.com/coder/coder/issues/2989 @BrunoQuaresma @dwahler  

### 📡 Tailscale Networking

Users have been reporting connection issues with WebRTC, and it’s been difficult to debug. We’ve had to build a lot of networking code that isn’t fun or easy to maintain. We can replace our WebRTC networking with Tailscale to remove ~3000 lines of networking code, and fix a bunch of issues for our users!

https://github.com/coder/coder/issues/2779 @kylecarbs

### 🌁 JetBrains Gateway

We lack docs for Gateway right now, which could give the perception that it doesn’t work with Coder. Ideally, someone that uses Gateway daily with Coder takes this (or someone that wants to, but there are too many bugs).

- Verify that Gateway works on the latest EAP when connecting to a Linux Coder agent. Working versions should be documented with a “last updated at” alongside the version. 
- Document the limitation of Gateway with non-Linux Coder agents (this is a limitation on their part, see:[https://youtrack.jetbrains.com/issue/CWM-3653/Remote-development-host-should-be-available-on-macOS-and-Windows](https://youtrack.jetbrains.com/issue/CWM-3653/Remote-development-host-should-be-available-on-macOS-and-Windows)). 

https://github.com/coder/coder/issues/3125 @spikecurtis 

### High priority bugs 
- #2960 

### 🚛 Nice to Haves

- [ ]  https://github.com/coder/coder/issues/3005
- [x]  https://github.com/coder/coder/issues/3025
- [ ]  https://github.com/coder/coder/issues/2764
- [ ]  https://github.com/coder/coder/issues/2752
- [ ]  https://github.com/coder/coder/issues/2970
- [ ]  https://github.com/coder/coder/issues/3000
- [x]  https://github.com/coder/coder/issues/3024
- [x]  https://github.com/coder/coder/issues/2714
- [x]  https://github.com/coder/coder/issues/1369
- [x]  https://github.com/coder/coder/issues/2836
- [x]  https://github.com/coder/coder/issues/2146
- [x]  https://github.com/coder/coder/issues/3013
- [ ]  https://github.com/coder/coder/issues/2966
- [x]  https://github.com/coder/coder/issues/2842
- [ ]  https://github.com/coder/coder/issues/2840
- [x]  https://github.com/coder/coder/issues/2969
- [x]  https://github.com/coder/coder/issues/3012
- [x]  https://github.com/coder/coder/issues/3031
- [ ]  https://github.com/coder/coder/issues/2815
- [x] #2188

**Comment by sharkymark** [comment]
@kylecarbs Health check for IDE icons ensures a good end-user experience. If the `startup_script` is still running, a user can click an IDE and it will fail.  Health check for IDE icons https://github.com/coder/coder/issues/2662

**Comment by kylecarbs** [comment]
Hmm, fair point. I wonder if we should adjust `startup_script` to potentially block the starting of an agent...? Seems like it could be a better approach.

**Comment by bpmct** [comment]
Some high-priority candidates for July or August:

- #2267
- #802 / #2660
- #696 
- #1763
- #1624
- #2760

Nice to have
- #2659
- #2262 / #2221 (better splash for marketing/demos)
- #3033
- #2918
- #3054

**Comment by kylecarbs** [comment]
We're taking out template.yaml and moving Helm chart into July @bpmct. #1763 will also be added. The rest will be in August.

**Comment by kylecarbs** [comment]
@bpmct I think you can swap in #2659, #2262, and #2221 for July. Just take some lesser important issues out of the nice to have (chefs choice).

**Comment by ammario** [comment]
For posterity, #2987 was added back since @sreya already began work.

Can we slot in https://github.com/coder/coder/issues/2135 since it's low effort and has been requested numerous times?

**Comment by ammario** [comment]
This is another severe bug https://github.com/coder/coder/issues/2960 related to workspace scheduling. I think it's more important than most of the current Nice to Haves.
